### ISSUE TITLE
Apply ArgumentNullException.ThrowIfNull to NLog source-code

### ISSUE THREAD

**Issue by snakefoot** [NLog/NLog.Web#868]
Has been introduced with NET6, but it should be "easy" to implement equal solution in NLog.Internal:

From:

```c#
            if (factory is null)
            {
                throw new ArgumentNullException(nameof(factory));
            }
```

To

```c#
           ArgumentNullException.ThrowIfNull(factory);
```

From:
```c#
            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
```

To:
```c#
           _logger = ArgumentNullException.ThrowIfNull(logger);
```

https://docs.microsoft.com/en-us/dotnet/api/system.argumentnullexception.throwifnull?view=net-6.0

Ex.

```c#
static internal class ArgumentNullException
{
    public static T ThrowIfNull<T>(T argument, [CallerArgumentExpression("argument")] string paramName = null)
        where T : class
    {
        if (argument is null)
        {
            throw new System.ArgumentNullException(paramName);
        }

        return argument;
    }
}
```

**Comment by bakgerman** [comment]
I have submit PR for this item

**Comment by snakefoot** [comment]
Resolved by #889
